src/python/dxpy/scripts/dx.py

```diff
@@ -3787,6 +3787,225 @@ def publish(args):
+            unarchive_requests[project_id].append({"folder": folder, "dryRun": args.dry_run, "rate": args.rate, "recurse":args.recurse})
+    
+    #TODO: confirm message
+    if not args.quiet:
+        if args.dry_run:
+            print('Dry run on unarchiving the following paths:')
+        else: 
+            print('Will try to unarchive the following paths:')
+        
+        for pid in unarchive_paths:
+            print("Project: [{}]: {}".format(dxpy.describe(pid)["name"],pid))
+            if unarchive_paths[pid]["files"]:
+                print(" -- File:")
+                for fid in unarchive_paths[pid]["files"]:
+                    fd = dxpy.describe(fid)
+                    print("     [{}]<{}>: {}".format(fd["name"], fd["archivalState"], fid))
+            if unarchive_paths[pid]["folders"]:    
+                print(" -- Folders:")
+                for fp in unarchive_paths[pid]["folders"]:
+                    print("     [{}]:".format(fp))
+                    for f in list(dxpy.find_data_objects(project=pid,
+                                               folder=fp,
+                                               classname="file",
+                                               recurse=args.recurse,
+                                               describe=True)):
+                        fdesc = f["describe"]
+                        print("             [{}/{}]<{}>: {}".format(fdesc["folder"], fdesc["name"], fdesc["archivalState"], fdesc["id"]))
+        
+        # print("\n".join("\t{}: {}".format(id, r)  for id, rs in unarchive_requests.items() for r in rs))
+        print()
+        
+    if args.confirm and INTERACTIVE_CLI and not args.dry_run:
+        if not prompt_for_yn('Confirm unarchiving all paths? ', default=True):
+            parser.exit(0)
+
+    
+    total = {"files": 0, "size": 0, "cost": 0}
+    dx_unarchive_errors = [InvalidState, ResourceNotFound, PermissionDenied]
+
+    for project_id in unarchive_requests:
+        for request_input in unarchive_requests[project_id]:
+            try:
+                res = dxpy.api.project_unarchive(project_id, request_input)
+                for k in res:
+                    total[k]+=res[k] 
+            except Exception as e:
+                print("Failed request: {}".format(request_input))
+                if type(e) in dx_unarchive_errors:
+                    print("     API error: {}. {}".format(e.name, e.msg))
+                else: 
+                    print("     Unexpected error: {}".format(format_exception(e)))
+                failed_path[project_id] = format_exception(e)
+                had_error = True
+                continue
+            if res["files"] == 0:
+                print("Failed request: {} is not unarchiving any file.".format(request_input))
+    
+    if not args.quiet:
+        if args.dry_run:
+            print(f'Would tag {total["files"]} files for unarchival, totalling {total["size"]} GB, costing {total["cost"]}')
```

Comment:
from spec <format the value returned by api call using project's currency setting>